quickjs.c

```diff
@@ -40185,237 +40189,265 @@ static JSValue js_iterator_from(JSContext *ctx, JSValue this_val,
     return JS_EXCEPTION;
 }
 
-static JSValue js_create_iterator_helper(JSContext *ctx, JSValue iterator,
-                                         JSIteratorHelperKindEnum kind, JSValue func, int64_t count);
-
-static JSValue js_iterator_proto_drop(JSContext *ctx, JSValue this_val,
-                                      int argc, JSValue *argv)
+static int check_iterator(JSContext *ctx, JSValue obj)
 {
-    JSValue v;
-    double dlimit;
-    int64_t limit;
-    if (!JS_IsObject(this_val))
-        return JS_ThrowTypeError(ctx, "Iterator.prototype.drop called on non-object");
-    v = JS_ToNumber(ctx, argv[0]);
-    if (JS_IsException(v))
-        return JS_EXCEPTION;
-    // Check for Infinity.
-    if (JS_ToFloat64(ctx, &dlimit, v)) {
-        JS_FreeValue(ctx, v);
-        return JS_EXCEPTION;
-    }
-    if (isnan(dlimit)) {
-        JS_FreeValue(ctx, v);
-        goto fail;
-    }
-    if (!isfinite(dlimit)) {
-        JS_FreeValue(ctx, v);
-        if (dlimit < 0)
-            goto fail;
-        else
-            limit = MAX_SAFE_INTEGER;
-    } else {
-        v = JS_ToIntegerFree(ctx, v);
-        if (JS_IsException(v))
-            return JS_EXCEPTION;
-        if (JS_ToInt64Free(ctx, &limit, v))
-            return JS_EXCEPTION;
-    }
-    if (limit < 0) {
-    fail:
-        return JS_ThrowRangeError(ctx, "must be positive");
+    if (!JS_IsObject(obj)) {
+        JS_ThrowTypeErrorNotAnObject(ctx);
+        return -1;
     }
-    return js_create_iterator_helper(ctx, this_val, JS_ITERATOR_HELPER_KIND_DROP, JS_UNDEFINED, limit);
+    return 0;
 }
 
-static JSValue js_iterator_proto_every(JSContext *ctx, JSValue this_val,
-                                       int argc, JSValue *argv)
+typedef struct JSIteratorHelperData {
+    JSValue obj;
+    JSValue next;
+    JSValue func; // predicate (filter) or mapper (flatMap, map)
+    JSValue inner; // innerValue (flatMap)
+    int64_t count; // limit (drop, take) or counter (filter, map, flatMap)
+    JSIteratorHelperKindEnum kind; 
```

Comment:
```suggestion
    JSIteratorHelperKindEnum kind : 8;
```
Otherwise it turns into an int and then the two bools in turn will take up 4 more bytes.